torch/csrc/jit/symbolic_script.cpp

```diff
@@ -949,6 +980,20 @@ c10::optional<GradientPair> gradientInfoForSchema(
     return cache_it->second;
   } else {
     auto schema_str = canonicalSchemaString(schema);
+    // Specialize Scalar to float for the arg type of the node schema
+    // this is used to:
+    // 1. define scalar type as float in TorchScript autodiff formula
+    // 2. to make sure the input of any graph node does not contain scalar type
+    //    in its argument, all scalar arg should already be passed with float
+    //    value since scalar/int aren't differentiable either way.
+    for (const Argument& arg : schema.arguments()) {
```

Comment:
Why does this need to iterate the schema arguments. It is doing a string replace below anyway.
Can't we just take the `schema_str` and globally find/replace Scalar with float. The intent is clearer and potential for bugs is clearer that way.